src/main/groovy/io/saagie/plugin/dataops/clients/SaagieClient.groovy

```diff
@@ -2313,4 +2393,104 @@ class SaagieClient {
+                if (parsedResultForAppList.data?.labWebApp) {
+                    listApps = parsedResultForAppList.data.labWebApp
+                }
+                return listApps
+            }
+        }, 'Unknown error in getAppListByProjectId', 'getAppListByProjectIdRequest Request')
+    }
+
+    String addAppVersion(app, appVersion) {
+        // 2. add appVersion id there is a appVersion config
+        if (appVersion?.exists()) {
+            Request addAppVersionRequest = saagieUtils.updateAppVersion(app?.id, appVersion)
+
+            client.newCall(addAppVersionRequest).execute().withCloseable { updateResponse ->
+                handleErrors(updateResponse)
+                String updateResponseBody = updateResponse.body().string()
+                def updatedAppVersion = slurper.parseText(updateResponseBody)
+                if (updatedAppVersion.data == null) {
+                    def message = "Something went wrong when adding new app version: $updateResponseBody"
+                    logger.error(message)
+                    throw new GradleException(message)
+                } else {
+                    String newAppVersion = updatedAppVersion.data.addJobVersion.number
+                    logger.info('Added new version: {}', newAppVersion)
+                    return newAppVersion
+                }
+            }
+        } else if (app?.id) {
+            Request listAppVersion = saagieUtils.getAppDetailRequest(app?.id)
+            client.newCall(listAppVersion).execute().withCloseable { listAppVersionsResponse ->
+                handleErrors(listAppVersionsResponse)
+                String listAppVersionResponseBody = listAppVersionsResponse.body().string()
+                def listAppVersionsData = slurper.parseText(listAppVersionResponseBody)
+                if (listAppVersionsData.data == null) {
+                    def message = "Something went wrong when getting list app versions: $listAppVersionResponseBody"
+                    logger.error(message)
+                    throw new GradleException(message)
+                } else {
+                    logger.info('getting list app versions: {}', listAppVersionsData.data.job.versions)
+                    String currentNumber
+                    listAppVersionsData.data.labWebApp.versions.each {
+                        if (it.isCurrent) {
+                            currentNumber = it.number
+                        }
+                    }
+                    return currentNumber
+                }
+            }
+        }
+    }
+
+
+    @Deprecated
+    String createProjectApp(App app, AppVersionDTO appVersion) {
+        logger.info('Starting deprecated createProjectApp task')
+        checkRequiredConfigForAppAndAppVersionAndProjectId(app, appVersion)
+
+
+        logger.debug('Using config [project={}, app={}, appVersion={}]', configuration.project, app, appVersion)
+
+        app.category=""
```

Comment:
You must note edit the property of a param, this will create a side effect ( we are trying to remove the side effects from the project every chance we get ). you need to change this property in the parent method that will use this method